data-prepper-core/src/main/java/com/amazon/dataprepper/pipeline/server/DataPrepperServer.java

```diff
@@ -12,13 +13,15 @@
 
     private final HttpServer server;
 
-    public DataPrepperServer(final int port) {
+    public DataPrepperServer(final int port, final DataPrepper dataPrepper) {
```

Comment:
Do we want to enforce non-null here (`Objects.requireNonNull`)? I'm asking because I saw this line below in a test file which threw me off a bit:
```
final DataPrepperServer dataPrepperServer = new DataPrepperServer(8080, null);
```

Could also be a question for the larger project itself - would these checks be useful in most constructors?

Reply:
Hmm, yeah. It's probably something that should be done at a larger level.

In this case, the constructor is just passing that reference to the constructors of some of the handlers, so maybe the NonNull requirement would make more sense in those constructors (because its not used explicitly here).